### PR TITLE
Changes to run the testsuite in the cloud

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Search for the public interface by testing multiple options.
In the cloud the names cannot be known.
The names depends on the instance_type and OS you selected.
Seems that not only debian like systems use "ens", but now also "centos7" in "t3" instances.
Additionally not interface "3" is selected, but "5" and "6".

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/features/profiles/Makefile
Comment (srbarrios) [inline testsuite/features/profiles/Makefile:14]: I'm not super happy having this temporary AWS hostnames in our code, a better approach I think it could be to import them from environment variables loaded in .bashrc on the controller.

File: testsuite/features/profiles/cloud_aws/Docker/Dockerfile
Comment (srbarrios) [inline testsuite/features/profiles/cloud_aws/Docker/Dockerfile:1]: Same worry than above, using temporary hostnames from AWS in out code.

File: testsuite/features/profiles/cloud_aws/Docker/authprofile/Dockerfile
Comment (srbarrios) [inline testsuite/features/profiles/cloud_aws/Docker/authprofile/Dockerfile:5]: Same

File: testsuite/features/profiles/cloud_aws/Docker/authprofile/sles15sp4.repo
Comment (srbarrios) [inline testsuite/features/profiles/cloud_aws/Docker/authprofile/sles15sp4.repo:5]: Can we access to that IBS repo from an AWS instance? 🤔

File: testsuite/features/support/twopence_init.rb
Comment (srbarrios) [inline testsuite/features/support/twopence_init.rb:337]: I have a doubt, if we run retail tests in AWS, we also need to adjust the next section where we set the private interface?

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh
Comment (srbarrios) [inline testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: Do we really need to still adding support for 4.1?
I think POS_Image-JeOS6_41 and JeOS7_41 is not anymore need.

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh
Comment (srbarrios) [inline testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh:27]: Not Uyuni but SUSE Manager (when Head)

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh
Comment (srbarrios) [inline testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh:1]: Do we miss profiles for 4.3 right?
I guess we need to start distinguishing between Head and 4.3, now that 4.3 is released.

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service
Comment (srbarrios) [inline testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service:2]: Not Uyuni but SUSE Manager, when applies.

File: testsuite/features/profiles/cloud_aws/Docker/Dockerfile
Comment (ktsamis) [inline testsuite/features/profiles/cloud_aws/Docker/Dockerfile:2]: this is deprecated according to the [docker doc](https://docs.docker.com/engine/reference/builder/#maintainer-deprecated), use something like LABEL org.opencontainers.image.authors="SvenDowideit@home.org.au".
Also use consistently the same email and name.

File: testsuite/features/profiles/cloud_aws/Docker/authprofile/Dockerfile
Comment (ktsamis) [inline testsuite/features/profiles/cloud_aws/Docker/authprofile/Dockerfile:6]: same as before, deprecated

File: testsuite/features/profiles/cloud_aws/Docker/serverhost/Dockerfile
Comment (ktsamis) [inline testsuite/features/profiles/cloud_aws/Docker/serverhost/Dockerfile:6]: same as before deprecated, also notice here that you use your email again, so the first one should be the same

**Review by ktsamis — CHANGES_REQUESTED** [review]
Small thing for consistency and to use the right value before it gets removed from docker completely and we start getting errors.

File: testsuite/features/profiles/cloud_aws/Docker/authprofile/sles15sp4.repo
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Docker/authprofile/sles15sp4.repo:5]: Hmm, good catch. We need to configure the mirror

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: This is autogenerated. Not really my area :-)

File: testsuite/features/profiles/Makefile
Comment (srbarrios) [reply testsuite/features/profiles/Makefile:14]: In fact, both current registry.mgr.suse.de (already in bashrc) and new AWS entries could be part of .bashrc on the controller. That will keep our test code a bit more decoupled from infrastructure pointers.
But yeah, feel free to merge that first PR, and maybe we can improve later. (Unless you find time to include it in sumaform).

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh
Comment (srbarrios) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: Maybe @aaannz knows about it?

Comment (aaannz) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: This is part of the retail images copied from [templates repo](https://github.com/SUSE/manager-build-profiles/tree/master/OSImage).
You, as QE, are deciding what to have in test suite, so if 4.1 is no longer tested, or will not be tested in AWS, there is no reason to have this copy there.

Comment (srbarrios) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: I don't see anything related to SUMA version in https://github.com/SUSE/manager-build-profiles/tree/master/OSImage, I mean there is no such folder suffixed with 41, 42, Head, Uyuni.
From where they come? I guess indeed, we can get rid of xxx_41 folder.

Comment (aaannz) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: templates repo does not differentiated between SUMA/Uyuni version. Templates work for all of them and what is build is provided by activation key.
However in test suite, because channels are not synced (at least they were not before), there is a fork of these templates with URL to external repositories included in the XML. Because repos URLs are added to XML, then you need different templates for 4.1, 4.2, etc.

File: testsuite/features/profiles/cloud_aws/Docker/Dockerfile
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Docker/Dockerfile:2]: This is generated and such changes should be made in the template which did not change.
I think this should go into a separate PR

File: testsuite/features/profiles/cloud_aws/Docker/authprofile/Dockerfile
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Docker/authprofile/Dockerfile:6]: This is generated and such changes should be made in the template which did not change.
I think this should go into a separate PR

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: I think the 4.1 testsuite also used the git URL to uyuni/master. So all testsuites (4.1, 4.2, 4.3, Head) use the templates from uyuni/master.
I think we should not remove them as long as 4.1 is still "supported".

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh:1]: I think this is not the PR where we should fix these things.

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh:25]: A good question for a different PR :-)

Comment (srbarrios) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh:25]: @aaannz @meaksh ? ^^^

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh
Comment (srbarrios) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh:1]: I don't get it, this is the PR to include 4.1 profile, but is not the PR to include 4.3 profile? :/

File: testsuite/features/profiles/Makefile
Comment (srbarrios) [inline testsuite/features/profiles/Makefile:7]:
```suggestion
DEFAULT_REPO = download.suse.de/ibs
```

**Review by srbarrios — CHANGES_REQUESTED** [review]
I think this PR is mixing 2 different goals and must be split in two PRs. This will simplify the review and possible requests.

1 - find public interface by testing multiple options
2 - Update Docker profiles to include AWS support

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh
Comment (aaannz) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh:25]: `images.sh` is called during image building process - https://osinside.github.io/kiwi/overview/workflow.html#description-components

Changing to salt bundle is not as simple as changing this call, but we pushed salt bundle capable initrd in 4.3.2. Changed templates are not yet published, waiting until release. See https://github.com/SUSE/manager-build-profiles/pull/33 for changes.

It would be great if we figure out some way to pass repos in different way then editing XML so you would be able to run very same templates our users use.

Comment (srbarrios) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh:25]: Has this something to be with the comment from @nadvornik here?
https://github.com/SUSE/spacewalk/issues/18970#issuecomment-1252044816

Comment (Bischoff) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/images.sh:25]: Rather than editing the XML at run time, you might also use a different profile for 4.3 than for 4.2. It might be simpler. I susepect other divergences will appear in the future.

File: testsuite/features/support/twopence_init.rb
Comment (mcalmer) [reply testsuite/features/support/twopence_init.rb:339]: Currently it seems that retail tests cannot be run in the cloud. So currently it does not create problems. But yes, potential we would need a detection as well.
But the detection in general is a bit weak. Just the lowest interface number is the "public" interface?

But currently it is the best we can do. Would it be ok, to use this for now and think about a better solution later?

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS6_41/config.sh:1]: ok, I will remove it from the cloud repo

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh:1]: testsuite/features/profiles/[Docker|Kiwi]/ are contains templates.
There is a "Makefile" which generate out of these templates the variants needed for running in Nuremberg, Provo or - now new - in the cloud.

The Makefile copy the templates into the new directories and replace some parts using `sed` to make the necessary changes. I just added a new target in the Makefile for cloud and run "make".

I did not change the template where this come from. To create a 4.3 profile, somebody should create it in the template folder and than just run "make".

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service:2]: This would need to be changed in the template. This is just a copy.

File: testsuite/features/profiles/Makefile
Comment (mcalmer) [reply testsuite/features/profiles/Makefile]: Just this is not enough. But I renamed REPO to DEFAULT_REPO

**Comment by mcalmer** [comment]
> I think this PR is mixing 2 different goals and must be split in two PRs. This will simplify the review and possible requests.
> 
> 1 - find public interface by testing multiple options
> 2 - Update Docker profiles to include AWS support

Well, I changed the title as all is required to run the testsuite in the cloud.
For testing I need "1" branch as I cannot run the testuite with multiple branches.

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh
Comment (srbarrios) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh:1]: I see. Ok, can you create a card for this purpose?
We need it to have this PR ported to Manager-4.3

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service
Comment (srbarrios) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service:2]: Can't we change manually this text after generating the profiles from the template?
It's a one time change.

**Comment by srbarrios** [comment]
> > I think this PR is mixing 2 different goals and must be split in two PRs. This will simplify the review and possible requests.
> > 1 - find public interface by testing multiple options
> > 2 - Update Docker profiles to include AWS support
> 
> Well, I changed the title as all is required to run the testsuite in the cloud. For testing I need "1" branch as I cannot run the testuite with multiple branches.

Ok, but I think that 1 could be already be merged in master.
And then you rebase.

File: testsuite/features/support/twopence_init.rb
Comment (srbarrios) [reply testsuite/features/support/twopence_init.rb:339]: If Retail is out of the scope for AWS, then let's keep that improvement for later yes.
In any case, I think a card for this, to remind it will be good.

File: testsuite/features/secondary/min_timezone.feature
Comment (srbarrios) [inline testsuite/features/secondary/min_timezone.feature:73]: That seems also out of the scope of this PR.

**Review by srbarrios — APPROVED** [review]
I'm gonna approve because I think we need to move on to other improvements, but personally I think we need to revisit this code later and refactor some parts.
My main worry is about using AWS hosts in some configuration files, instead of having these values coming from an environment variable in the controller.

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/config.sh:1]: Done.
https://github.com/SUSE/spacewalk/issues/19147

I am a bit unsure who will get this. I added it to Planning

File: testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service
Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service:2]: You can write a "sed" command to the Makefile. Otherwise it would be overwritten when the next one call "make"

Comment (mcalmer) [reply testsuite/features/profiles/cloud_aws/Kiwi/POS_Image-JeOS7_head/root/etc/systemd/system/image-deployed.service:2]: ahh, no. It is easier. We just change the template

File: testsuite/features/support/twopence_init.rb
Comment (mcalmer) [reply testsuite/features/support/twopence_init.rb:339]: https://github.com/SUSE/spacewalk/issues/19148

**Pull request merged**
